### PR TITLE
feat: Add (internal) mock "Ext" logger implementation for testing

### PR DESCRIPTION
**Pull request by Josmithr**
Adds a mock implementation of the `ITelemetryLoggerExt` interface for use in testing. Exported (internal-only) as a minimal interface and a factory function so as to avoid redundant implementation details and to avoid exporting any (new) classes.

### CODE REVIEW

**Comment by Josmithr** [comment]
Needed by #21463

File: packages/utils/telemetry-utils/src/mockLogger.ts [hunk 1]
```diff
@@ -259,3 +263,32 @@ function matchObjects(actual: ITelemetryPropertiesExt, expected: ITelemetryPrope
...
+}
+
+/**
+ * Creates an {@link IMockLoggerExt}.
+ *
+ * @internal
+ */
+export function createMockLoggerExt(minLogLevel?: LogLevel): IMockLoggerExt {
+	const mockLogger = new MockLogger(minLogLevel);
+	const childLogger = createChildLogger({ logger: mockLogger });
```
Comment (alexvy86) [inline packages/utils/telemetry-utils/src/mockLogger.ts:288]: Found while looking through the class for review.
```suggestion
	const childLogger = mockLogger.toTelemetryLogger();
```

File: packages/utils/telemetry-utils/src/mockLogger.ts [hunk 2]
```diff
@@ -259,3 +263,32 @@ function matchObjects(actual: ITelemetryPropertiesExt, expected: ITelemetryPrope
 	}
 	return true;
 }
+
+/**
+ * Mock {@link ITelemetryLoggerExt} implementation.
+ *
+ * @remarks Can be created via {@link IMockLoggerExt}.
```
Comment (alexvy86) [inline packages/utils/telemetry-utils/src/mockLogger.ts:270]:
```suggestion
 * @remarks Can be created via {@link createMockLoggerExt}.
```

**Review by alexvy86 — APPROVED** [review]
I'm a bit unsure about the need for the interface once MockLogger has readonly-events, so I'd consider leaving a TODO if we want to get rid of it at that point, but I'm not actively opposed to it existing either.

File: packages/utils/telemetry-utils/src/mockLogger.ts (hunk 1 above)
Comment (Josmithr) [reply packages/utils/telemetry-utils/src/mockLogger.ts:288]: I considered doing it this way, but I honestly think that method should be removed. The "ext" types conceptually extend the base types, so I find it weird (and smelly) to expose knowledge about the "ext" types from a base type implementation.

There is a lot about `MockTelemetryLogger` that I think needs to be reworked. It has a few bugs, exposes state mutability, etc.

I think I'm going to leave this one alone for now, unless you feel strongly.

Comment (Josmithr) [reply packages/utils/telemetry-utils/src/mockLogger.ts:288]: Also posted this in Teams about removing this class from our (non-internal) exports: https://teams.microsoft.com/l/message/19:50292e8934024fc19d6ca2080dd7681e@thread.skype/1718659268797?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=9ce27575-2f82-4689-abdb-bcff07e8063b&parentMessageId=1718659268797&teamName=Fluid%20Framework&channelName=Dev&createdTime=1718659268797

Comment (alexvy86) [reply packages/utils/telemetry-utils/src/mockLogger.ts:288]: Nope, go ahead 👍.

For future consideration, I'd say we only need one mock implementation of a logger class which can implement both interfaces, so I guess I don't really think of it as just "a base type implementation". The Ext types ended up being alpha right now, on their way to internal, but I'd be fine having this class implement both and also end up being internal. I'm not sure if we have partners using it right now, and if we do it should be trivial for them to copy/create their own mock base logger implementation.

Comment (Josmithr) [reply packages/utils/telemetry-utils/src/mockLogger.ts:288]: Also, a work-in-progress PR where I'm attempting to make `events` readonly.
https://github.com/microsoft/FluidFramework/pull/21470

**Comment by Josmithr** [comment]
> I'm a bit unsure about the need for the interface once MockLogger has readonly-events, so I'd consider leaving a TODO if we want to get rid of it at that point, but I'm not actively opposed to it existing either.

Their `events` properties are exposed as two different types, unfortunately. 1 is `ITelemetryEventBase` and the other is `ITelemetryEventExt`. With some refactoring, we could probably capture both with a single interface and a generic type parameter, but that would require re-working some things more substantially.

**Pull request merged**